### PR TITLE
chore: fix flaky group skipping core e2e

### PR DESCRIPTION
**Pull request by WeizhengSap**
ticket: CXSPA-1628

### CODE REVIEW

**Review by bgambocjaviniar — CHANGES_REQUESTED** [review]
This PR made checkout more flaky than before

If you take a look at these, it failed 3 times in a few instances
1) https://dashboard.cypress.io/projects/k3nmep/runs/a31945ec-e4f1-4441-8651-e1b83c3a869e/test-results/4a482c8c-de28-44e2-9741-2cacc76d6722 (b2b checkout is failing now (it wasn't before) -> it couldn't go through the payment step)
2) https://dashboard.cypress.io/projects/k3nmep/runs/407ea9d9-cb2e-4da9-be10-b554985c1a23/test-results/95bf0e4e-aa33-4108-a6d7-32266f44edf4 (group skipping is still failing -> it couldn't go through to the payment step as well)
3) https://dashboard.cypress.io/projects/k3nmep/runs/35006/overview/00f7cd1a-a8fd-4871-bddb-2474c38f42a0 (group skipping is still failing -> it couldn't go through to the payment step as well)

---
You can view these instances in the dashboard from the filter [here](https://dashboard.cypress.io/projects/k3nmep/runs?branches=%5B%7B%22label%22%3A%22feature%2FCXSPA-1628%22%2C%22suggested%22%3Afalse%2C%22value%22%3A%22feature%2FCXSPA-1628%22%7D%5D&committers=%5B%5D&flaky=%5B%5D&page=1&status=%5B%5D&tags=%5B%5D&timeRange=%7B%22startDate%22%3A%221970-01-01%22%2C%22endDate%22%3A%222038-01-19%22%7D) 

How I did to test these changes?
Ran multiple builds to run the e2es (each build are unique so unique e2e answers as well) 
https://sap-cx.slack.com/archives/GDQNTBPU2/p1666706308379939

File: projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-flow.ts [hunk 1]
```diff
@@ -202,26 +202,27 @@ export function fillAddressForm(shippingAddressData: AddressData = user) {
...
-  const deliveryPage = waitForPage(
-    '/checkout/delivery-mode',
-    'getDeliveryPage'
-  );
-  fillShippingAddress(shippingAddressData);
-  cy.wait(`@${deliveryPage}`).its('response.statusCode').should('eq', 200);
-
   cy.wait('@putDeliveryMode').its('response.statusCode').should('eq', 200);
+
+  const getCheckoutDetailsAlias = interceptCheckoutB2CDetailsEndpoint();
```
Comment (bgambocjaviniar) [inline projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-flow.ts:225]: Not a problem putting it here, but I believe intercepts should always be defined at the top level than as a next statement

File: projects/storefrontapp-e2e-cypress/cypress/integration/accessibility/group-skipping.core-e2e-spec.ts [hunk 2]
```diff
@@ -37,16 +37,8 @@ context('Group Skipping - Checkout', () => {
       checkout.goToProductDetailsPage();
       checkout.addProductToCart();
 
-      cy.intercept({
```
Comment (bgambocjaviniar) [inline projects/storefrontapp-e2e-cypress/cypress/integration/accessibility/group-skipping.core-e2e-spec.ts:40]: removing this made group skipping more apparent as a flaky e2e as seen in the dashboard

**Review by bgambocjaviniar — CHANGES_REQUESTED** [review]
overall, I suggest to undo everything and most likely wait for something before it tries to click to the next step (to the payment method step).

looking at the video https://dashboard.cypress.io/projects/k3nmep/runs/34606/test-results/6d87d7b4-2811-46a7-bdf3-29846d932b16/video, I believe the 'criteria' to go to the next step was not complete even if we have that disabled button in place before clicking. Maybe there's an extra step it needs to wait for to make sure we can go to the next step.

All assumptions is being made, but again, if you test before these changes, it is unlikely that checkout fails, but with these changes, checkout fails often.

To re-iterate, I would suggest to maybe wait for 'something' to go to payment method, or try with a 2s static wait before we click that button to go to the payment method step (this is to test using the magic button and running like ~15 builds parallely)

File: projects/storefrontapp-e2e-cypress/cypress/integration/accessibility/group-skipping.core-e2e-spec.ts (hunk 2 above)
Comment (WeizhengSap) [reply projects/storefrontapp-e2e-cypress/cypress/integration/accessibility/group-skipping.core-e2e-spec.ts:40]: This `putDeliveryMode` is also checked/wait in function `fillAddressForm`. So, this check is duplicate.

File: projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-flow.ts (hunk 1 above)
Comment (WeizhengSap) [reply projects/storefrontapp-e2e-cypress/cypress/helpers/checkout-flow.ts:225]: Spy early has a risk that this request was done before we want to check. For example, there may have multiple `getCheckoutDetailsAlias`.  We want to check the one that sent after `putDeliveryMode`.

**Comment by WeizhengSap** [comment]
After creating a new address for the cart, it is no need to set this address to cart. In the failed e2e tests, I can see that the "set address to cart" has 400 error (bad request). 

Screen Shot 2022-11-10 at 3 15 45 PM

So, I fixed this "create + set" address issue in checkout code.

**Comment by WeizhengSap** [comment]
From "delivery mode" step to "payment" step, if the `@putDeliveryMode` is done, and `cy.wait(`@${getCheckoutDetailsAlias}`)` is done, nothing should block the payment page render after cms paged loaded.

**Review by bgambocjaviniar — APPROVED** [review]

**Pull request merged**